### PR TITLE
Apply bettertransformer in convert.py

### PR DESCRIPTION
**Pull request by slyalin**
BetterTransformers transformation enables ScaledDotProductAttention operation in exported models. Disabled by default, to enable use `--bettertransformer` in `convert.py`.

### CODE REVIEW

**Comment by slyalin** [comment]
@eaidova, tried to set `use_cache=True` in the `model.config` before export to OV, but it didn't help.

File: llm_bench/python/convert.py [hunk 1]
```diff
@@ -346,33 +345,100 @@ def convert_causal_lm(args):
...
-    ov_out_dir = Path(args.output_dir) / 'pytorch/dldt' / precision
-    save_tokenizer(tok, ov_out_dir)
+        model = model.to_bettertransformer()
 
-    start1 = time.perf_counter()
-    model.save_pretrained(ov_out_dir)
-    end1 = time.perf_counter()
-    log.info(f'Serialization total time {end1 - start1}s')
+        onnx_config_constructor = TasksManager.get_exporter_config_constructor(model=model, exporter="onnx", task="text-generation")
+        config = onnx_config_constructor(model.config, use_past=True)
```
Comment (eaidova) [inline llm_bench/python/convert.py:358]:
```suggestion
        config = onnx_config_constructor(model.config, use_past=True, use_past_in_inputs=True)
```

File: llm_bench/python/convert.py [hunk 2]
```diff
@@ -346,33 +345,99 @@ def convert_causal_lm(args):
...
+
+        def pathed_generate_dummy_inputs(self, *args, **kwargs):
+            dummy_inputs = self._original_generate_dummy_inputs(*args, **kwargs)
+            if 'input_ids' in dummy_inputs and dummy_inputs['input_ids'].shape[1] == 1:
+                dummy_inputs['input_ids'] = torch.cat([dummy_inputs['input_ids'], dummy_inputs['input_ids']], dim=-1)
+                attention_mask = dummy_inputs['attention_mask']
+                dummy_inputs['attention_mask'] = torch.cat([attention_mask, attention_mask.new_ones((attention_mask.shape[0], 1))], dim=-1)
+            return dummy_inputs
+
+        # also as we are not using intel-optimimum model class for save/store the model we also need to patch inputs to
```
Comment (eaidova) [inline llm_bench/python/convert.py:371]: Could you please check, I think it does not required anymore with new optimum version
https://github.com/huggingface/optimum/blob/main/optimum/exporters/onnx/config.py#L97

File: llm_bench/python/convert.py [hunk 3]
```diff
@@ -346,33 +345,99 @@ def convert_causal_lm(args):
...
-    model = OVModelForCausalLM.from_pretrained(
-        args.model_id,
-        export=True,
-        compile=False,
-        trust_remote_code=True,
-        load_in_8bit=False,
-        config=AutoConfig.from_pretrained(args.model_id, trust_remote_code=True),
-    )
-    end = time.perf_counter()
+    if args.bettertransformer:
```
Comment (eaidova) [inline llm_bench/python/convert.py:348]: I think for consolidation all conversion code in one place and avoid duplication, you can reuse https://github.com/openvinotoolkit/openvino.genai/blob/eccde42b3ea39802e01e050a7b0a7cec97254e19/llm_bench/python/convert.py#L243

it replicate original optimum flow with allowing configs overriding and patching models that are not part of optimum. All things model getting onnx config, conversion, saving configs, weights compression are already present on it, so it maybe simpler to integrate bettertransformers inside.

File: llm_bench/python/convert.py [hunk 4]
```diff
@@ -240,10 +240,32 @@ def patch_model_for_optimum_export(model):
...
+    model_config.generate_dummy_inputs = types.MethodType(pathed_generate_dummy_inputs, model_config)
+
+    return model
+
+
 def convert_optimum_causallm_base(model, args):
     tok = AutoTokenizer.from_pretrained(args.model_id, trust_remote_code=True)
     model = patch_model_for_optimum_export(model)
     model_config = model.config
+
```
Comment (slyalin) [inline llm_bench/python/convert.py:268]:
```suggestion
```

**Review by eaidova — APPROVED** [review]

**Pull request merged**